### PR TITLE
T2viz ux improvements

### PR DESCRIPTION
**Pull request by ruanyl**
### Description
1. Allow to customize the AI assistant logo via config
2. Display a more precise message when the generate PPL returns no result
3. Only pre-select index pattern that is not disabled
4. When no index pattern is selected, disabled nature language input and the submit button

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test.
- [ ] New functionality has user manual doc added.
- [ ] Commits are signed per the DCO using --signoff.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/ui_action_context_menu.tsx [hunk 1]
```diff
@@ -31,6 +31,7 @@ export const ActionContextMenu = (props: Props) => {
           trigger: AI_ASSISTANT_QUERY_EDITOR_TRIGGER as any,
         })),
         closeMenu: () => setOpen(false),
+        title: '',
```
Comment (Hailong-am) [inline public/components/ui_action_context_menu.tsx:34]: does this empty title is needed here?

File: public/components/visualization/text2vega.ts [hunk 2]
```diff
@@ -63,6 +63,9 @@ export class Text2Vega {
                 )
                 // eslint-disable-next-line @typescript-eslint/no-explicit-any
                 .toPromise<any>();
+              if (res.rawResponse.total === 0) {
+                throw new Error(`There is no result with the generated query: '${value.ppl}'.`);
+              }
```
Comment (Hailong-am) [inline public/components/visualization/text2vega.ts:68]: would there a place to show this error on page?

File: public/components/visualization/source_selector.tsx [hunk 3]
```diff
@@ -52,13 +54,20 @@ export const SourceSelector = ({
...
       !selectedSourceId &&
       dataSourceOptions.length > 0 &&
       dataSourceOptions[0].options.length > 0
     ) {
-      onChange(dataSourceOptions[0].options[0]);
+      const options = dataSourceOptions[0].options;
+      const selectedOption = options.find((o) => !o.disabled);
+      if (selectedOption) {
+        onChangeRef.current(selectedOption);
+      }
```
Comment (Hailong-am) [inline public/components/visualization/source_selector.tsx:70]: I can't find where `onChangeRef` been consumed

**Review by Hailong-am — APPROVED** [review]

File: public/components/ui_action_context_menu.tsx (hunk 1 above)
Comment (ruanyl) [reply public/components/ui_action_context_menu.tsx:34]: Yes, the default title is `Options` but we don't need a title, so set it to `''`

File: public/components/visualization/text2vega.ts (hunk 2 above)
Comment (ruanyl) [reply public/components/visualization/text2vega.ts:68]: Yes, in [text2viz.tsx](https://github.com/opensearch-project/dashboards-assistant/pull/330/files#diff-44cdcb0c0e12c1b4082203f1a1030c5a257a8cec28587574f3e43811eadf1dbb) line 110

File: public/components/visualization/text2viz.tsx [hunk 4]
```diff
@@ -342,16 +343,17 @@ export const Text2Viz = () => {
             onChange={(e) => setInput(e.target.value)}
             fullWidth
             compressed
-            prepend={<EuiIcon type={chatIcon} />}
+            prepend={<EuiIcon type={config.branding.logo || chatIcon} />}
```
[the remark this answers is not in the document]
Comment (ruanyl) [reply public/components/visualization/text2viz.tsx:346]: yes, url to svg is supported

**Pull request merged**